### ISSUE TITLE
Some single-driver builds fail due to non-ASCII characters in comments

### ISSUE THREAD

**Issue by rzero9** [mamedev/mame#5482]
When making certain single-driver configurations, GENIe fails with this output. The example is for fmtowns.cpp, but it also fails with other drivers (e.g. at.cpp, amstrad.cpp)

```
[MINGW64] C:\msys64\src\mame>make SOURCES=src\mame\drivers\fmtowns.cpp SUBTARGET=fmtowns VERBOSE=1
GCC 9.1.0 detected
mkdir "build\\generated\\mame\\layout\\" 2> nul || exit 0
mkdir "build\\generated\\mame\\fmtowns\\" 2> nul || exit 0
mkdir "build\\generated\\mame\\drivers\\" 2> nul || exit 0
mkdir "build\\generated\\mame\\machine\\" 2> nul || exit 0
3rdparty/genie/bin/windows/genie.exe --distro=generic --OPTIMIZE=3 --target='mame' --subtarget='fmtowns' --build-dir='build' --PYTHON_EXECUTABLE='C:\msys64\mingw64\bin\python.exe' --SOURCES='src\mame\drivers\fmtowns.cpp' --osd='windows' --targetos='windows' --PLATFORM='x86' --gcc=mingw64-gcc --gcc_version=9.1.0 gmake
Traceback (most recent call last):
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 211, in <module>
    parse_file(root,filename,'')
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 162, in parse_file
    parse_file(root, fullname, newfolder)
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 166, in parse_file
    parse_file(root, fullname.replace('.h','.cpp'), newfolder)
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 120, in parse_file
    for line in fp.readlines():
  File "C:\msys64\mingw64\lib\python3.7\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 488: character maps to <undefined>
Traceback (most recent call last):
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 211, in <module>
    parse_file(root,filename,'')
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 162, in parse_file
    parse_file(root, fullname, newfolder)
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 166, in parse_file
    parse_file(root, fullname.replace('.h','.cpp'), newfolder)
  File "c:/msys64/src/mame/scripts/build/makedep.py", line 120, in parse_file
    for line in fp.readlines():
  File "C:\msys64\mingw64\lib\python3.7\encodings\cp1252.py", line 23, in decode
    return codecs.charmap_decode(input,self.errors,decoding_table)[0]
UnicodeDecodeError: 'charmap' codec can't decode byte 0x90 in position 488: character maps to <undefined>
c:/msys64/src/mame/scripts/genie.lua:1555: attempt to call a nil value
stack traceback:
        c:/msys64/src/mame/scripts/genie.lua:1555: in main chunk
        [C]: in upvalue 'builtin_dofile'
        [string "premake = { }..."]:84: in function 'dofile'
        [string "_WORKING_DIR        = os.getcwd()..."]:45: in function '_premake_main'
makefile:1055: recipe for target 'build/projects/windows/mamefmtowns/gmake-mingw64-gcc/Makefile' failed
make: *** [build/projects/windows/mamefmtowns/gmake-mingw64-gcc/Makefile] Error 1
```

This is on Windows Server 2019 (functionally equivalent to Windows 10 1809) with the standard build tools. I haven't tested on Mac or Linux.

This seems to be caused by non-ASCII characters in the comments of several source files:

src\devices\bus\rs232\rs232.cpp
src\devices\bus\sunkbd\hlekbd.cpp
src\devices\machine\ie15_kbd.cpp

Removing those comments works as a workaround.

**Comment by happppp** [comment]
It's because makedep.py is incompatible with the newest version of Python 3. To make it work, change makdep line 64 to:
fp = open(root + srcfile, 'r', encoding="utf-8")

line 115 to:
fp = open(root + srcfile, 'r', encoding="utf-8")

**Comment by drencorxeen** [comment]
Is this workaround going to be made pushed as part of the repo?

**Comment by happppp** [comment]
It would break the build on python 2.x, I'm not a Python programmer at all, maybe someone can come up with a solution that works on both versions.

**Comment by cuavas** [comment]
Addressed in 9a6e04b01f830d8491e1c98eabf19015f817c852
